.github/workflows/build-and-test.yml

```diff
@@ -4,6 +4,8 @@ on:
     branches:
       - main
       - develop
+  pull_request:
+    types: [opened, synchronize, reopened]
```

Comment:
These are actually [the default types](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request), but I specified them for clarity since there are a lot of potential event types that may fire related to pull requests.